src/main/python/ttconv/imsc/attributes.py

```diff
@@ -297,6 +298,31 @@ class TemporalAttributeParsingContext:
   frame_rate: Fraction = Fraction(30, 1)
   tick_rate: int = 1
 
+@dataclass
+class TemporalAttributeWritingContext:
+  frame_rate: Fraction = Fraction(30, 1)
+  tick_rate: int = 1
+
+def to_time_format(context: TemporalAttributeWritingContext, time: Fraction) -> str:
+  if context.frame_rate.numerator == 1 and context.frame_rate.denominator == 1:
+    return to_hh_mm_ss_ms(time)
+  return to_frames(context, time)
+
+def to_hh_mm_ss_ms(time: Fraction) -> str:
+  value = time.numerator / time.denominator
+  value = datetime.timedelta(seconds = value)
+  #millsec = value.microseconds / 1000
+  #value = str(datetime.timedelta(seconds = value)) 
+
+  return f"{value}s"
+
+def to_frames(context: TemporalAttributeWritingContext, time: Fraction) -> str:
+  value = time.numerator / time.denominator
+  value *= (context.frame_rate.numerator / context.frame_rate.denominator)
```

Comment:
Currently truncating. This is likely the correct behavior as if you are not on a frame boundary, you are in a frame.

Reply:
`Fraction` instance should be multiplied, so that precision can be preserved.

See https://www.w3.org/TR/ttml-imsc1.1/#synchronization for trunc vs ceiling

Reply:
Done